src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -303,18 +319,13 @@ FixedwingPositionControl::calculate_target_airspeed(float airspeed_demand, const
 
 	// groundspeed undershoot
 	if (!_l1_control.circle_mode()) {
-
-		// rotate ground speed vector with current attitude
-		Vector2f yaw_vector(_R_nb(0, 0), _R_nb(1, 0));
-		yaw_vector.normalize();
-
-		const float ground_speed_body = yaw_vector * ground_speed;
-
 		/*
 		 * This error value ensures that a plane (as long as its throttle capability is
 		 * not exceeded) travels towards a waypoint (and is not pushed more and more away
 		 * by wind). Not countering this would lead to a fly-away.
 		 */
+		const float ground_speed_body = _body_velocity(0);
```

Comment:
TODO: review, this isn't right

Reply:
@dagar What's not right about this? Isn't it the same logic as before?

Reply:
The underspeed behaviour when you start climbing/descending.